### ISSUE TITLE
Milestone #4 Tasks

### ISSUE THREAD

**Issue by sweber15** [UBC-MDS/DSCI522_group315#40]
**Milestone 4**
- Sam 
- Subing
- [x]  Dockerfile

- [x] DockerHub upload

Sarah
- [x]  Makefile dependency graph
- [x] Clean up make file
- [x] Post dependency on readme 

**Feedback**
Sam:
- [x] Makefile summary should have a slightly more detailed description of what the file does - a sentence instead of just "ufc judging analysis data pipe"

- [x] Peer feedback: "Include ROC curve, f1 score etc." either address this, mention in issue why you don't think it makes sense to address this, or mention this as something to do in your future directions and then leave the issue open.

- [x] Add a caption for feature list 

- [x] Figure 1 add a label on the bottom with the features

Sarah:

- [x] Change Figure 3 plot to validation error and add text to plot instead of the legend

- [x] Table 1: give the features more readable labels

- [x] Table 2: re-shape so that you have columns "training accuracy" and "testing accuracy" for rows "no feature selection" and "selected features"

- [x] Confusion matrix make the labels for numbers bigger

Subing:
- [x]  Add a second test at the end of download to ensure it's actually been downloaded

- [x] Remove duplication in list: "The criteria sets priority from high to low ranking effective Striking/Grappling, effective Striking/Grappling and cage/ring Control(The ABC MMA Rules Committee 2017)." (this was noted also in Milestone 2)

- [x] Address this peer feedback: "Does it make sense to have RFE? The judges see and should consider all of these factors." Do this in the issue and link back here. 

- [x] Figure 2: Create an appendix for the all but the first image 

- [x] Figure 2: Change jitter box plot to split violin graph

Last Priority:
- [ ] Would be nice to functionalize more (#34) but the commenting does make it easy to read in the current state

**Comment by SamEdwardes** [comment]
Built initial Dockerfile. See #41. Will probably require a few updates once we finalize the Make to graph bit.

**Comment by SamEdwardes** [comment]
> Figure 1 add a label on the bottom with the features

See #47.

> Add a caption for feature list

See PR #48

> Makefile summary should have a slightly more detailed description of what the file does - a sentence instead of just "ufc judging analysis data pipe"

See PR #48

> Peer feedback: "Include ROC curve, f1 score etc." either address this, mention in issue why youon't think it makes sense to address this, or mention this as something to do in your future directions and then leave the issue open.

See https://github.com/UBC-MDS/DSCI522_group315/issues/32#issuecomment-583162218

**Comment by sweber15** [comment]
PR #49  Changes:
 

- Added Makefile dependency graph 
- rearranged Makefile to show the dependencies better

     - Note: I tried to use data/02_preprocessed/* instead of X_train, y_train, etc. However, when I ran make all it ran the script four times. 

**Still need to post on readme, but once it is merged I want to play around with size**

ML Script Steps Completed:
- Changed Figure 3 plot to validation error and train error as annotations on the plot instead of as a legend.
- Table 1: Gave the features more readable labels using the feature chart from the Data section of the report. 
- Table 2: re-shaped to have columns for "training accuracy" and "testing accuracy"  and "no feature selection" and "selected features" for rows. 
- Made text bigger in the confusion matrix

Additional:
- Re-uploaded figures and csvs that were changed
- Changed sections of the report related to the changes in the script

**Comment by sweber15** [comment]
Regarding our last priority "would be nice to functionalize more (#34 ) but the commenting does make it easy to read in the current state".

We agree it is a nice to have instead of a need to have for this project in the given timeframe. The EDA is a good example of having functions in the script. If we had more time to refine the code this would definitely be something we would do especially for the ML Script. However, we are satisfied with the successfully complete of our scripts for our analysis and the documentation currently in place is easy to follow and replicate. This is something we would consider for an additional version.
